src/Compilers/CSharp/Portable/Symbols/TypeWithState.cs

```diff
@@ -19,13 +18,17 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
 
         public static TypeWithState ForType(TypeSymbol type)
         {
-            var state = type?.CanContainNull() != false ? NullableFlowState.MaybeNull : NullableFlowState.NotNull;
-            return new TypeWithState(type, state);
+            return Create(type, NullableFlowState.MaybeDefault);
         }
 
         public static TypeWithState Create(TypeSymbol type, NullableFlowState defaultState)
         {
-            var state = defaultState == NullableFlowState.MaybeNull && type?.CanContainNull() != false ? NullableFlowState.MaybeNull : NullableFlowState.NotNull;
+            if (defaultState == NullableFlowState.MaybeDefault &&
+                (type is null || type.IsTypeParameterDisallowingAnnotation()))
```

Comment:
>IsTypeParameterDisallowingAnnotation [](start = 38, length = 36)

I think the correct test is slightly more complex.  See comments elsewhere.

Reply:
I've not found a test where that change is observable. For now, I've added an assert.

---
In reply to: [347695398](https://github.com/dotnet/roslyn/pull/39778#discussion_r347695398) [](ancestors = 347695398)